### PR TITLE
Add docs about auto detection of issue tracker

### LINKED ISSUE #58: Add docs about the issue tracker auto detection based on git config
Add some docs about this feature to the README.md

Related issue: #50

### PR DESCRIPTION
**Pull request by mehdy**
Closes #58

### CODE REVIEW

**Review by preslavmihaylov — CHANGES_REQUESTED** [review]
Could you also document this in the [Github](https://github.com/preslavmihaylov/todocheck/blob/master/README.md#github) and [Gitlab](https://github.com/preslavmihaylov/todocheck/blob/master/README.md#gitlab) sections?

File: README.md [hunk 1]
```diff
@@ -72,10 +72,11 @@ Place the binary in a folder, shined upon by your `$PATH`.
  * For macos & linux, that's typically `/usr/local/bin/`
 
 # Quickstart
-You will need a `.todocheck.yaml` configuration file in the root of your project first.  
+For public github/gitlab projects it should detect the issue tracker from your git configuration.
```
[the remark this answers is not in the document]
Comment (mehdy) [reply README.md:75]: Absolutely :see_no_evil:

File: README.md [hunk 2]
```diff
@@ -125,7 +127,8 @@ The first time you run the application, it will ask for your [personal access to
 After you've specified it, it will store it in the auth tokens cache for subsequent executions. See the [Authentication](#authentication) section for more info.
 
 ## [Gitlab](https://gitlab.com)
-To integrate with a public gitlab repository, specify the origin of your repo and the `GITLAB` issue tracker in your `.todocheck.yaml` configuration:
+To integrate with a public github repository, there's no need to provide a `.todocheck.yaml` explicitly as it can automatically detect the issue tracker based on the git remote address.
```
Comment (preslavmihaylov) [inline README.md:130]: gitlab, not github

File: README.md [hunk 3]
```diff
@@ -125,7 +127,8 @@ The first time you run the application, it will ask for your [personal access to
...
+And if you want to explicit, please specify the origin of your repo and the `GITLAB` issue tracker in your `.todocheck.yaml` configuration:
```
Comment (preslavmihaylov) [inline README.md:131]: And if you want to **be** explicit

File: README.md [hunk 4]
```diff
@@ -105,7 +106,8 @@ Currently, todocheck supports the following issue trackers:
 | [Redmine](https://redmine.org/)                 | Supports public access with no auth & private access via an API token |
 
 ## [Github](https://github.com)
-To integrate with a public github repository, specify the origin of your repo and the `GITHUB` issue tracker in your `.todocheck.yaml` configuration:
+To integrate with a public github repository, there's no need to provide a `.todocheck.yaml` explicitly as it can automatically detect the issue tracker based on the git remote address.
+And if you want to explicit, please specify the origin of your repo and the `GITHUB` issue tracker in your `.todocheck.yaml` configuration:
```
Comment (preslavmihaylov) [inline README.md:110]: Same comments here as for the gitlab part

File: README.md [hunk 5]
```diff
@@ -72,10 +72,11 @@ Place the binary in a folder, shined upon by your `$PATH`.
  * For macos & linux, that's typically `/usr/local/bin/`
 
 # Quickstart
-You will need a `.todocheck.yaml` configuration file in the root of your project first.  
+For public github/gitlab projects it will detect the issue tracker from your git configuration.
+Otherwise you will need a `.todocheck.yaml` configuration file in the root of your project first.
```
Comment (preslavmihaylov) [inline README.md:76]: This line assumes some context although it's in the start of the `Quickstart` section. This section will typically be read by newcomers to the tool.

For example, it already mentions an issue tracker when it isn't already mentioned that todocheck needs a way to integrate with one.

I suggest restructuring this part in the following way:
```
First, you need to configure `todocheck`'s integration with your issue tracker.
This is done by creating a `.todocheck.yaml` file in the root of your project.

For public github or gitlab repositories, you need not specify a config file, as long as you are using git, as `todocheck` will interpret your issue tracker from the git configuration.

Alternatively, the explicit configuration for a public github repository would look like this:
(config follows)
```

**Review by preslavmihaylov — CHANGES_REQUESTED** [review]
There are still some minor typos & I've also suggested restructuring the `Quickstart` section to be more friendly to first-time readers.

**Comment by mehdy** [comment]
> There are still some minor typos & I've also suggested restructuring the `Quickstart` section to be more friendly to first-time readers.

Good points. Thanks! I'll make the changes.

**Review by preslavmihaylov — APPROVED** [review]
All looks good now!

**Comment by preslavmihaylov** [comment]
Feel free to squash & merge unless you intend to make any more changes.

**Pull request merged**